backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/preloader/PreloaderBundleGenerator.java

```diff
@@ -116,17 +122,21 @@ public String generate (TreeLogger logger, GeneratorContext context, String type
 		}
 
 		for (Entry<String, ArrayList<Asset>> bundle : bundles.entrySet()) {
-			StringBuffer buffer = new StringBuffer();
+			StringBuilder buffer = new StringBuilder();
 			for (Asset asset : bundle.getValue()) {
-				String path = asset.file.path().replace('\\', '/').replace(assetOutputPath, "").replaceFirst("assets/", "");
-				if (path.startsWith("/")) path = path.substring(1);
+				String pathOrig = asset.filePathOrig.replace('\\', '/').replace(assetOutputPath, "").replaceFirst("assets/", "");
+				if (pathOrig.startsWith("/")) pathOrig = pathOrig.substring(1);
+				String pathMd5 = asset.fileMd5.path().replace('\\', '/').replace(assetOutputPath, "").replaceFirst("assets/", "");
+				if (pathMd5.startsWith("/")) pathMd5 = pathMd5.substring(1);
 				buffer.append(asset.type.code);
 				buffer.append(":");
-				buffer.append(path);
+				buffer.append(pathOrig);
+				buffer.append(":");
+				buffer.append(pathMd5);
 				buffer.append(":");
-				buffer.append(asset.file.isDirectory() ? 0 : asset.file.length());
+				buffer.append(asset.fileMd5.isDirectory() ? 0 : asset.fileMd5.length());
```

Comment:
Looks like a mistake on first glance, but it is just the naming that is confusing and should be improved.